alias.c

```diff
@@ -681,17 +648,21 @@ bool mutt_addr_is_user(struct Address *addr)
   return false;
 }
 
-void mutt_alias_free(struct Alias **p)
+void mutt_alias_free(struct Alias *p)
 {
-  struct Alias *t = NULL;
+  if (!p)
+    return;
+
+  mutt_alias_delete_reverse(p);
+  FREE(&p->name);
+  mutt_addr_free(&p->addr);
+  FREE(&p);
+}
 
-  while (*p)
+void mutt_alias_free_all() {
```

Comment:
I'd prefer if this took the AliasList to be freed.
There's only one, atm, but that may change.